utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -499,5 +503,183 @@ void PessimisticTransactionDB::UnregisterTransaction(Transaction* txn) {
   transactions_.erase(it);
 }
 
+// Returns true if commit_seq <= snapshot_seq
+bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
```

Comment:
One question is are we going to assume there are external synchronization? Like `VersionSet::last_sequence_` provide a global fence, where we can assume all operation by commits with commit_seq <= snapshot_seq happen before the creation of the snapshot. I'm not sure if we should make this assumption. If we can make this assumption, some of the locking (or memory_order) can be relaxed.